docs/index.bs

```diff
@@ -3608,6 +3740,29 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
           <li>Set <var>registration</var> to the result of running <a href="#scope-match-algorithm">Match Service Worker Registration</a> algorithm passing <var>request</var>'s <a for="request">url</a> as the argument.</li>
           <li>If <var>registration</var> is null or <var>registration</var>'s <a href="#dfn-active-worker">active worker</a> is null, return null.</li>
           <li>Set <var>client</var>'s <a href="#dfn-service-worker-client-active-worker">active worker</a> to <var>registration</var>'s <a href="#dfn-active-worker">active worker</a>.</li>
+          <li>If <var>request</var> is a <a>navigation request</a> and <var>registration</var>'s <a>navigation preload enabled flag</a> is set, and <var>request</var>'s <a>method</a> is `<code>GET</code>`, then:
+            <ol>
+              <li>Let <var>preloadRequest</var> be the result of <a>cloning the request</a> <var>request</var></li>
+              <li>Let <var>preloadRequestHeaders</var> be <var>preloadRequest</var>'s <a>header list</a></li>
+              <li>Let <var>preloadResponseObject</var> be a new {{Response}} object and a new associated {{Headers}} object whose <a>guard</a> is "<code>immutable</code>"</li>
+              <li><a>Append to header list</a> <var>preloadRequestHeaders</var> a new <a>header</a> whose <a>name</a> is `<code>Service-Worker-Navigation-Preload</code>` and <a>value</a> is <var>registration</var>'s <a>navigation preload header value</a></li>
+              <li>Set <var>preloadRequest</var>'s <a>skip-service-worker flag</a></li>
+              <li>Run the following substeps <a>in parallel</a>:
+                <ol>
+                  <li>
+                    <p><a>Fetch</a> <var>preloadRequest</var>.</p>
+                    <p>To <a>process response</a> for <var>navigationPreloadResponse</var>, run these substeps:</p>
```

Comment:
I was using https://fetch.spec.whatwg.org/#fetch-method has a guide, and *response* seems to come out of nowhere there too. What's the right thing to do here, and why is fetch right to do it this way?